src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/constructors.jl

```diff
@@ -217,17 +217,13 @@ end
 ####################################################
 
 function Base.:+(ac1::RationalEquivalenceClass, ac2::RationalEquivalenceClass)
-    if toric_variety(ac1) !== toric_variety(ac2)
-        throw(ArgumentError("The rational equivalence classes must be defined on the same toric variety, i.e. the same OSCAR variable"))
-    end
+    @req toric_variety(ac1) === toric_variety(ac2) "The rational equivalence classes must be defined on the same toric variety"
     return RationalEquivalenceClass(toric_variety(ac1), polynomial(ac1) + polynomial(ac2))
 end
```

Comment:
@fingolfin Regarding the `OSCAR variable`: The key question here is what `same` means. In theory, we could say it is sufficient if `toric_variety(ac1)` and `toric_variety(ac2)` are isomorphic, for then we could apply this isomorphism say to `ac1` and then compare the result with `ac2`.

In practice, we do not do this. (Among others because it can be challenging to find such an isomorphism.) For an illustrative example, consider the following:
```julia
P2a = projective_space(NormalToricVariety, 2)
P2b = projective_space(NormalToricVariety, 2)
```
Then we find that `P2a == P2b` (as well as `P2a === P2b`) both return `false`. And so, besides `P2a` and `P2b` being isomorphic via the identity, we (at least as of now) cannot compare rational equivalence classes on `P2a` with those on `P2b`.

Rather, for addition, subtraction, comparison etc. of two rational equivalence classes, they would both have to be defined on `P2a` (or both on `P2b`). We tried to describe this fact by saying that the rational equivalence classes would have to be defined on the "same OSCAR variable" `P2a` (respectively `P2b`). Is there a better formulation that comes to your mind? We can of course also add a description, similar to this message. What do you think?

(This also applies mutually in a couple of other places, e.g. for toric divisors.)

Reply:
Is the wording "blabla must be equal" ambiguous? Would people equate this with being isomorphic?

Reply:
I understand why you used "same OSCAR variable" now, but I still think it's confusing and doesn't make sense (no offense intended).

I don't think anyone would expect "isomorphic" to be meant by "same". But yes, someone might think that two rings that "look the same" are "the same"... Though personally "same" for me carries a strong connotation of "identical" (i.e., `===`), which is more than "equal" (i.e., `==`).

So to me, the message as it is now is perfectly clear, but of course some people will read "same" as "equal" and then might be confused... Perhaps using the word "identical" would help them; perhaps not. Using "the same OSCAR variable" does not help, though, because it (a) still uses "same" (so if you think it means `==` or even "are isomorphic" the first time around, why wouldn't you think the same the second time around?), and (b) "OSCAR variable" is a non-existing concept, and even "Julia variable" is wrong. Perhaps "must be identical objects"  would be helpful?


All that said: In the end, no matter what you do, you can't stop people from bashing their head against this. Even if you write "must be identical as in '===' must return true for both objects" some people will either (a) not read the error and just complain that they don't understand the error, or (b) complain that this is an unjust restricting and the comparison should be changed to "==". This is a problem that is not specific to math either, it affects most people who do non-trivial programming at some point.

Reply:
We are not writing software for us, rather for mathematicians. So yes, they will not understand this. They will probably already be confused by `!(P2a == P2b)`. I thought that the sentence "i.e. the same OSCAR/Julia variable" at least gives a hint that this is not mathematical equality. At least with that sentence they will be confused for the right reason.

Reply:
I agree: With this explanation in mind, the error message is now perfectly clear. So let us change it as you proposed @fingolfin . Thank you!

Reply:
I like "must be identical blabla" (this is also what the docstring for `===` uses).

Alternatively and with more CS lingo: "must be egal".

Reply:
@lkastner but there is no variable involved. I don't see how that could be illuminating to anyone? If the only thing it does is to give a hint that something "non-mathematical" happens (and for the record, I strongly disagree with this!), then we should rather spell it out ("here identical refers to the programming concept, this is stricter than mathematical equality"). But that seems cumbersome.

Anyway: I am happy to have this use `identical` consistently. But that affects many places, not just those touched by this PR, and possibly in other repos, too, so I'd prefer to not do it in this PR.